### ISSUE TITLE
open slim.report.json: permission denied with version 1.36

### ISSUE THREAD

**Issue by beltran-rubo** [docker-slim/docker-slim#201]
Expected Behavior
=================
 No error

---
Actual Behavior
=================

Using docker-slim version linux|Transformer|1.35.0|6eca3ad331720c31a7e398ffaf002a896dc9a70a|2021-04-15_02:48:03AM -> works well

Using docker-slim version linux|Transformer|1.36.0|5dad378523ed9d53f6e5de21f6b98f07da4f7d59|2021-06-13_02:46:04AM -> ERROR below

time="2021-06-14T16:40:34Z" level=fatal msg="docker-slim: failure" error="open slim.report.json: permission denied" stack="goroutine 1 [running]:\nruntime/debug.Stack(0x4, 0x40d730, 0xc0001e3d20)\n\truntime/debug/stack.go:24 +0x9f\ngithub.com/docker-slim/docker-slim/pkg/util/errutil.FailOn(0xf014a0, 0xc00061e480)\n\tgithub.com/docker-slim/docker-slim/pkg/util/errutil/errutil.go:28 +0x54\ngithub.com/docker-slim/docker-slim/pkg/report.(*Command).saveInfo(0xc000142e00, 0xd0eaa0, 0xc000142e00, 0xc00062a1e0)\n\tgithub.com/docker-slim/docker-slim/pkg/report/command_report.go:429 +0x26b\ngithub.com/docker-slim/docker-slim/pkg/report.(*BuildCommand).Save(...)\n\tgithub.com/docker-slim/docker-slim/pkg/report/command_report.go:443\ngithub.com/docker-slim/docker-slim/pkg/app/master/commands/build.OnCommand(0xc00013e008, 0xc000134230, 0x7ffe1d452819, 0x10, 0x0, 0xc000164000, 0xc000159d00, 0x14d08b8, 0x0, 0x0, ...)\n\tgithub.com/docker-slim/docker-slim/pkg/app/master/commands/build/handler.go:890 +0x510b\ngithub.com/docker-slim/docker-slim/pkg/app/master/commands/build.glob..func1(0xc000144160, 0xc000148000, 0xc000148540)\n\tgithub.com/docker-slim/docker-slim/pkg/app/master/commands/build/cli.go:515 +0x27fc\ngithub.com/urfave/cli.HandleAction(0xca4aa0, 0xe21d70, 0xc000144160, 0xc000144160, 0x0)\n\tgithub.com/urfave/cli@v1.22.4/app.go:526 +0xfd\ngithub.com/urfave/cli.Command.Run(0xddeb95, 0x5, 0x0, 0x0, 0x145a9b0, 0x1, 0x1, 0xe161cb, 0x6c, 0x0, ...)\n\tgithub.com/urfave/cli@v1.22.4/command.go:174 +0x58e\ngithub.com/urfave/cli.(*App).Run(0xc00012e000, 0xc000030300, 0x2e, 0x30, 0x0, 0x0)\n\tgithub.com/urfave/cli@v1.22.4/app.go:279 +0x7e8\ngithub.com/docker-slim/docker-slim/pkg/app/master.Run()\n\tgithub.com/docker-slim/docker-slim/pkg/app/master/app.go:15 +0x54\nmain.main()\n\tgithub.com/docker-slim/docker-slim/cmd/docker-slim/main.go:8 +0x25\n" version="linux|Transformer|1.36.0|5dad378523ed9d53f6e5de21f6b98f07da4f7d59|2021-06-13_02:46:04AM"

---
Steps to Reproduce the Problem
===============================

Run a 'docker-slim build' with different versions: 1.35 and 1.36. I am using the binaries directly from GitHub releases (not compiled myself).

---
Specifications
=================
  - Version:  version linux|Transformer|1.36.0|5dad378523ed9d53f6e5de21f6b98f07da4f7d59|2021-06-13_02:46:04AM 
  - Platform:  Linux x86_64

**Comment by kcq** [comment]
@beltran-rubo thank you for opening the issue! Do you mind sharing a bit more about your setup? Where do you have the `docker-slim` executable and where do you run your `docker-slim` command? 

The `slim.report.json` files is the the report file for the command you run and it gets saved in the current working directory where you are when you run the `docker-slim` command. Saving the report file may fail if you are somewhere where your current system user doesn't have permissions to write.

**Comment by beltran-rubo** [comment]
I have the 'docker-slim' & 'docker-slim-sensor' binaries in my home directly: "~/docker-slim build ...".

```
-rwxr-xr-x   1 beltran beltran     7806976 Jun 14 16:37  docker-slim-sensor
-rwxr-xr-x   1 beltran beltran    17367040 Jun 14 16:37  docker-slim
-rw-r--r--   1 beltran beltran       20151 Jun 11 16:29  slim.report.json
```

It fails if I execute the 'docker-slim' command from a directory with root privileges: 'cd /var && ~/docker-slim build ...'

**Comment by kcq** [comment]
@beltran-rubo Thanks for the additional context! `slim.report.json` gets saved in the current working directory by default. You can pick a different location using the global `--report` flag like this: `docker-slim --report your/custom/path/to/slim.report.json build your-image`.

In the next release I'll add an enhancement to use the temp directory if saving to the current directory fails.

**Comment by kcq** [comment]
@beltran-rubo  the 1.36.1 version should have the command report location enhancement... do you mind trying it - https://github.com/docker-slim/docker-slim/releases/tag/1.36.1

**Comment by beltran-rubo** [comment]
Thanks for the quick turnaround, I can confirm this is fixed in 1.36.1.

**Comment by kcq** [comment]
nice!
